sympy/series/formal.py

```diff
@@ -1143,6 +1144,192 @@ def integrate(self, x=None, **kwargs):
+
+        """
+
+        k = self.ak.variables[0]
+        ak_seq = sequence(self.ak.formula, (k, 1, oo))
+        fact_seq = sequence(factorial(k), (k, 1, oo))
+        bell_seq = (ak_seq * fact_seq)
+        inner_coeffs = []
+        for j in range(1, n+1):
+            inner_coeffs.append(bell(n, j, tuple(bell_seq[:n-j+1])))
+
+        k = Dummy('k')
+        return sequence(tuple(inner_coeffs), (k, 1, oo))
+
+    def compose(self, other, x=None, n=6):
+        r"""
+        Returns the truncated terms of the formal power series of the composed function,
+        up to specified `n`.
+
+        If `f` and `g` are two formal power series of two different functions,
+        then the coefficient sequence ``ak`` of the composed formal power series
+        `fp` will be :-
+
+        .. math::
+        \sum\limits_{k=0}^{n} b_k B_{n,k}(x_1, x_2, \dotsc, x_{n-k+1})
+
+        Parameters
+        ==========
+
+        n : Number, optional
+            Specifies the order of the term up to which the polynomial should
+            be truncated.
+
+        Examples
+        ========
+
+        >>> from sympy import fps, sin, exp, bell
+        >>> from sympy.abc import x
+        >>> f1 = fps(exp(x))
+        >>> f2 = fps(sin(x))
+
+        >>> f1.compose(f2, x)
+        1 + x + x**2/2 - x**4/8 - x**5/15 + O(x**6)
+
+        >>> f1.compose(f2, x, n=8)
+        1 + x + x**2/2 - x**4/8 - x**5/15 - x**6/240 + x**7/90 + O(x**8)
+
+        See Also
+        ========
+
+        sympy.functions.combinatorial.numbers.bell
+
+        References
+        ==========
+
+        .. [1] Comtet, Louis: Advanced combinatorics; the art of finite and infinite expansions. Reidel, 1974.
+
+        """
+
+        if x is None:
+            x = self.x
+        if n is None:
+            return iter(self)
+
+        other = sympify(other)
+
+        if not isinstance(other, FormalPowerSeries):
+            raise ValueError("Both series should be an instance of FormalPowerSeries"
+                             " class.")
+
+        if self.dir != other.dir:
+            raise ValueError("Both series should be calculated from the"
+                             " same direction.")
+        elif self.x0 != other.x0:
+            raise ValueError("Both series should be calculated about the"
+                             " same point.")
+        elif self.x != other.x:
+            raise ValueError("Both series should have the same symbol.")
+
+        f, g = self.function, other.function
+        if other._eval_term(0).as_coeff_mul(other.x)[0] is not S.Zero:
+            raise ValueError("The formal power series of the inner function should not have any "
+                "constant coefficient term.")
+
+        k = Dummy('k')
+        terms = []
+
+        for i in range(1, n):
+            bell_seq = other.coeff_bell(i)
+            fact_seq = sequence(factorial(k), (k, 1, oo))
+            seq = (self.ak * fact_seq * bell_seq)[:i]
```

Comment:
I would suggest saving the infinite version of ``seq``. That way all you have to do is slice at computation time. Since ``sequences`` are lazy saving them shouldn't decrease performance.

Reply:
This can be done at the time of defining the classes. And moreover, how do we save `bell_seq`? That is dependent on `i`, and also on a function defined in fps.